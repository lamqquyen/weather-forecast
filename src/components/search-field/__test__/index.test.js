import React from "react";
import SearchField from "../index";
import TestRenderer from "react-test-renderer";
import {FormControl} from "react-bootstrap";

describe("SearchField", () => {
  it("renders correctly with loading is false", () => {
    const tree = TestRenderer.create(<SearchField />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with loading is true", () => {
    const tree = TestRenderer.create(<SearchField loading={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with focusing on form input", () => {
    const tree = TestRenderer.create(<SearchField value={'test'}/>).root;
    const {act} = TestRenderer

    const formControl = tree.findByType(FormControl);

    act(() => {
      formControl.props.onFocus()
    })

    const suggestionBlock = tree.findByProps({className: 'search-suggestion__wrapper'})

    expect(suggestionBlock).not.toBeNull()
  });
});
