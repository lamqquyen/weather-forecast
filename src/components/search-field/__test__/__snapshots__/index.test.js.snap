// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders correctly 1`] = `
<div
  className="input-group"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
      id="basic-addon1"
    >
      <i
        className="fa fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    value=""
  />
</div>
`;

exports[`SearchField renders correctly with loading is true 1`] = `
<div
  className="input-group"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
      id="basic-addon1"
    >
      <i
        className="fa fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    value=""
  />
</div>
`;
